sim/scripts/synth_fpga.tcl

```diff
@@ -0,0 +1,132 @@
+# See LICENSE for license details.
+
+
+# Amazon FPGA Hardware Development Kit
+#
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Amazon Software License (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+#    http://aws.amazon.com/asl/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+################################################################################
+# Command-line Arguments
+################################################################################
+
+set strategy                [lindex $argv  0]
+set desired_host_frequency  [lindex $argv  1]
+set output                  [lindex $argv  2]
+set firesim_xdc             [lindex $argv  3]
+set firesim_sv              [lindex $argv  4]
+
+set device_type         "xcvu9p-flgb2104-2-i"
+set uram_option         2
+
+################################################################################
+# Read design files
+################################################################################
+
+if {$uram_option != 2} {
+  set_param synth.elaboration.rodinMoreOptions {rt::set_parameter disableOregPackingUram true}
+  set_param physynth.ultraRAMOptOutput false
+}
+
+# Maintain DONT TOUCH functionality for 2020.2 onwards
+if {[string match *2020.2* [version -short]] || [string match *2021.* [version -short]]} {
+  set_param project.replaceDontTouchWithKeepHierarchySoft false
+}
+
+switch $uram_option {
+    "2" {
+        set synth_uram_option "-max_uram_cascade_height 2"
+        set uramHeight 2
+    }
+    "3" {
+        set synth_uram_option "-max_uram_cascade_height 3"
+        set uramHeight 3
+    }
+    "4" {
+        set synth_uram_option "-max_uram_cascade_height 1"
+        set uramHeight 4
+    }
+    default {
+        set synth_uram_option "-max_uram_cascade_height 1"
+        set uramHeight 4
+    }
+}
+
+switch $strategy {
+    "BASIC" {
```

Comment:
Could you somehow pull in these values from the underlying strategies in AWS-FPGA?

Reply:
I'd rather keep these separate.

Reply:
We'll want to tweak these options and it's easier to do if they're in this repository.

Reply:
Also, I want this to be a bit target-independent in the hopes of finding vivado messups.